pilot/helpers/Project.py

```diff
@@ -212,88 +214,44 @@ def save_file(self, data):
         Args:
             data: { name: 'hello.py', path: 'path/to/hello.py', content: 'print("Hello!")' }
         """
-        # TODO fix this in prompts
-        if 'path' not in data:
-            data['path'] = data['name']
-
-        if 'name' not in data or data['name'] == '':
-            data['name'] = os.path.basename(data['path'])
-        elif not data['path'].endswith(data['name']):
-            if data['path'] == '':
-                data['path'] = data['name']
-            else:
-                data['path'] = data['path'] + '/' + data['name']
-        # TODO END
+        name = data['name'] if 'name' in data and data['name'] != '' else os.path.basename(data['path'])
+        path = data['path'] if 'path' in data else name
 
-        data['path'], data['full_path'] = self.get_full_file_path(data['path'], data['name'])
-        update_file(data['full_path'], data['content'])
+        path, full_path = self.get_full_file_path(path, name)
+        update_file(full_path, data['content'])
 
-        (File.insert(app=self.app, path=data['path'], name=data['name'], full_path=data['full_path'])
+        (File.insert(app=self.app, path=path, name=name, full_path=full_path)
          .on_conflict(
             conflict_target=[File.app, File.name, File.path],
             preserve=[],
-            update={'name': data['name'], 'path': data['path'], 'full_path': data['full_path']})
+            update={'name': name, 'path': path, 'full_path': full_path})
          .execute())
 
     def get_full_file_path(self, file_path: str, file_name: str) -> Tuple[str, str]:
```

Comment:
this function is not working properly for many test cases on mac, lets remove changes made to this function from this PR and do it in separate PR just for this function refactoring once it is finished